apps/base-docs/tutorials/docs/6_subquery.md

```diff
@@ -0,0 +1,364 @@
+
+Mapping functions define how blockchain data is transformed into the optimised GraphQL entities that we previously defined in the `schema.graphql` file.
+
+Navigate to the default mapping function in the `src/mappings` directory. You will be able to see two exported functions `handleNftClaim` and `handleDailyAggregation`:
+
+```ts
+export async function handleNftClaim(log: TokensClaimedLog): Promise<void> {
+  logger.info(`New claim log at block ${log.blockNumber}`);
+  assert(log.args, "No log.args");
+
+  let date = new Date(Number(log.block.timestamp) * 1000);
+
+  const claim = Claim.create({
+    id: log.transactionHash,
+    blockHeight: BigInt(log.blockNumber),
+    timestamp: date,
+    claimer: log.args.claimer,
+    receiver: log.args.receiver,
+    tokenId: log.args.startTokenId.toBigInt(),
+    quantity: log.args.quantityClaimed.toBigInt(),
+  });
+
+  await handleDailyAggregation(date, claim.quantity);
+
+  await claim.save();
+}
+
+export async function handleDailyAggregation(
+  date: Date,
+  quantity: bigint,
+): Promise<void> {
+  const id = date.toISOString().slice(0, 10);
+  let aggregation = await DailyAggregation.get(id);
+  logger.info(`New daily aggregation at ${id}`);
+
+  if (!aggregation) {
+    aggregation = DailyAggregation.create({
+      id,
+      totalQuantity: BigInt(0),
+    });
+  }
+
+  aggregation.totalQuantity = aggregation.totalQuantity + quantity;
+
+  await aggregation.save();
+}
+```
+
+The `handleNftClaim` function receives a `log` parameter of type `TokensClaimedLog` which includes log data in the payload. We extract this data and then save this to the store using the `.save()` function (_Note that SubQuery will automatically save this to the database_).
+
+
+### Build
+
+Next, build your work to run your new SubQuery project. Run the build command from the project's root directory as given here:
+
+```shell
+yarn build
+```
+
+:::warning
+Whenever you make changes to your mapping functions, you must rebuild your project.
+:::
+
+## Run & Query
+
+Now, you are ready to run your first SubQuery project. Let’s check out the process of running your project in detail.
+
+### Run Locally
+
+The `docker-compose.yml` file defines all the configurations that control how a SubQuery node runs. For a new project, which you have just initialised, you won't need to change anything.
+
+However, visit the [Running SubQuery Locally](../../run_publish/run.md) to get more information on the file and the settings.
+
+Run the following command under the project directory:
+
+```shell
+yarn start:docker
+```
+
+:::info
+It may take a few minutes to download the required images and start the various nodes and Postgres databases.
+:::
+
+### Query 
+
+Next, let's query our project. Follow these three simple steps to query your SubQuery project:
+
+1. Open your browser and head to `http://localhost:3000`.
+
+2. You will see a GraphQL playground in the browser and the schemas which are ready to query.
```

Comment:
Please use `1.` for every numbered item.  That way, the numbers will get automatically assigned and we won't have to worry about it when/if the items in the list change.